test/IRGen/temporary_allocation/codegen.swift

```diff
@@ -66,32 +66,42 @@ withUnsafeTemporaryAllocation(of: Void.self, capacity: 2) { buffer in
 // CHECK: [[VOID_PTR:%[0-9]+]] = ptrtoint [2 x i8]* [[VOID_PTR_RAW]] to [[WORD]]
 // CHECK: call swiftcc void @blackHole([[WORD]] [[VOID_PTR]])
 
-// MARK: Very large allocation
+// MARK: Large allocation
 
 // A large allocation size should produce an OS version check, call to
 // swift_stdlib_isStackAllocationSafe(), and then a branch based on the result
 // to either stack-allocate or heap-allocate.
-withUnsafeTemporaryAllocation(byteCount: 0x0FFF_FFFF, alignment: 1) { buffer in
+withUnsafeTemporaryAllocation(byteCount: 0xFFFF, alignment: 1) { buffer in
   blackHole(buffer.baseAddress)
 }
 // CHECK-apple: [[IS_OS_OK:%[0-9]+]] = call swiftcc i1 @"$ss26_stdlib_isOSVersionAtLeastyBi1_Bw_BwBwtF"
 // CHECK-apple: br i1 [[IS_OS_OK]], label %[[OS_OK_BR:[0-9]+]], label %[[UNSAFE_BR:[0-9]+]]
 
 // CHECK-apple: [[UNSAFE_BR]]:
 // CHECK-unknown: [[UNSAFE_BR:[0-9]+]]:
-// CHECK: [[HEAP_PTR_RAW:%[0-9]+]] = call noalias i8* @swift_slowAlloc([[WORD]] 268435455, [[WORD]] -1)
+// CHECK: [[HEAP_PTR_RAW:%[0-9]+]] = call noalias i8* @swift_slowAlloc([[WORD]] 65535, [[WORD]] -1)
 // CHECK: [[HEAP_PTR:%[0-9]+]] = ptrtoint i8* [[HEAP_PTR_RAW]] to [[WORD]]
 // CHECK: call swiftcc void @blackHole([[WORD]] [[HEAP_PTR]])
 // CHECK: call void @swift_slowDealloc(i8* [[HEAP_PTR_RAW]], [[WORD]] -1, [[WORD]] -1)
 
 // CHECK-apple: [[OS_OK_BR]]:
-// CHECK: [[IS_SAFE:%[0-9]+]] = call zeroext i1 @swift_stdlib_isStackAllocationSafe([[WORD]] 268435455, [[WORD]] 1)
+// CHECK: [[IS_SAFE:%[0-9]+]] = call zeroext i1 @swift_stdlib_isStackAllocationSafe([[WORD]] 65535, [[WORD]] 1)
 // CHECK: br i1 [[IS_SAFE]], label %[[SAFE_BR:[0-9]+]], label %[[UNSAFE_BR]]
 
 // CHECK: [[SAFE_BR]]:
 // CHECK: [[SPSAVE:%spsave[0-9]*]] = call i8* @llvm.stacksave()
-// CHECK: [[STACK_PTR_RAW:%temp_alloc[0-9]*]] = alloca [268435455 x i8], align 1
-// CHECK: [[STACK_PTR:%[0-9]+]] = ptrtoint [268435455 x i8]* [[STACK_PTR_RAW]] to [[WORD]]
+// CHECK: [[STACK_PTR_RAW:%temp_alloc[0-9]*]] = alloca [65535 x i8], align 1
+// CHECK: [[STACK_PTR:%[0-9]+]] = ptrtoint [65535 x i8]* [[STACK_PTR_RAW]] to [[WORD]]
 // CHECK: call swiftcc void @blackHole([[WORD]] [[STACK_PTR]])
 // CHECK: call void @llvm.stackrestore(i8* [[SPSAVE]])
 
+// MARK: Excessively large allocation
+
+// An excessively large allocation should always end up on the heap.
+withUnsafeTemporaryAllocation(byteCount: 0xFFFF_FFFF, alignment: 1) { buffer in
```

Comment:
Isn't the idea that this should _never_ call `swift_stdlib_isStackAllocationSafe`? Perhaps it's worth checking that.

Reply:
Right, that's what I said. Shouldn't this test include a `CHECK-NOT` for the call to `swift_stdlib_isStackAllocationSafe`?

Reply:
Ah! Sure, I can add that. :)